### PR TITLE
Throw a globalEvent when a tag is created. 

### LINKED ISSUE #3521: Add Tagging Event
When tagging something trigger an event to which mission makers can subscribe. Probably passing the tag, position, color, object as parameters.

Pinging random people: @BaerMitUmlaut @esteldunedain

### PR DESCRIPTION
**Pull request by esteldunedain**
**When merged this pull request will:**
- Throw a globalEvent when a tag is created. Useful for mission makers.
- Close #3521.

Parameters are:
```
0: Tag object, which can be used to get the position, orientation, etc <OBJECT>
1: Color ("black", "red", "green", "blue") <STRING>
2: Object that was tagged (objNull if it was the floor) <OBJECT>
3: Unit that created the tag <OBJECT>
```

@bux578, will you handle documentation?

### CODE REVIEW

**Comment by bux578** [comment]
Will do

File: addons/tagging/functions/fnc_createTag.sqf [hunk 1]
```diff
@@ -19,8 +20,8 @@
 
 #include "script_component.hpp"
 
-params ["_tagPosASL", "_vectorDirAndUp", "_color", "_object"];
-TRACE_4("createTag:", _tagPosASL, _vectorDirAndUp, _color, _object);
+params ["_tagPosASL", "_vectorDirAndUp", "_color", "_object", "_unit"];
+TRACE_5("createTag:", _tagPosASL, _vectorDirAndUp, _color, _object, _unit);
 
 if !((toLower _color) in ["black", "red", "green", "blue"]) exitWith {
     ACE_LOGERROR_1("%1 is not a valid tag colour.", _color);
```
Comment (bux578) [inline addons/tagging/functions/fnc_createTag.sqf]: `colour.", _color`

örgh

File: addons/tagging/functions/fnc_createTag.sqf [hunk 2]
```diff
@@ -31,6 +32,9 @@ _tag setObjectTextureGlobal [0, '\z\ace\addons\tagging\UI\tags\' + _color + '\'
 _tag setPosASL _tagPosASL;
 _tag setVectorDirAndUp _vectorDirAndUp;
 
+// Throw a global event for mision makers
+["tagCreated", [_tag, _color, _object, _unit]] call EFUNC(common,globalEvent);
```
Comment (bux578) [inline addons/tagging/functions/fnc_createTag.sqf]: Do we have a convention on how to name events?
Maybe QGVAR it?

Comment (jonpas) [inline addons/tagging/functions/fnc_createTag.sqf]: We don't yet, but we should do QGVAR.

Comment (esteldunedain) [inline addons/tagging/functions/fnc_createTag.sqf]: Yes, QGVAR is something we'll have to do, specially if we switch to CBA events.

The bad thing is that we are not doing it almost anywhere atm.

I'll change it here though

Comment (jonpas) [inline addons/tagging/functions/fnc_createTag.sqf]: It will require some time to get it moved, we should do it the "deprecation" way like with functions.

**Comment by esteldunedain** [comment]
@bux578, event renamed to "ace_tagging_tagCreated"

**Comment by Glowbal** [comment]
Why do we all of a sudden have an ACE prefix for this event? I don't think any of the other events we use have such a prefix. We are not yet using a CBA event framework that I know off, so I do not see the reasons to prefix this event.

If you are going with "future proof it", in case we do switch over to a system were we do require to prefix all our events, every third party has to double check any event used anyway. No benefit in doing it now.

**Comment by esteldunedain** [comment]
> I don't think any of the other events we use have such a prefix.

Missile_Guidance, Explosives and Mk6 use prefixed event names. The rest of modules don't.

I'm fine either way. Let's discuss it this afternoon/evening.

**Comment by esteldunedain** [comment]
The event name is back to being unprefixed, for consistency with most other current event names.

**Pull request merged**